### PR TITLE
basic struct support

### PR DESCRIPTION
**Pull request by vandrada**
Added encoding support for structs

### CODE REVIEW

**Comment by vandrada** [comment]
Sorry, accidentally closed this issue on OctroDroid

**Comment by cblage** [comment]
Looks good! Thanks :D

File: lib/json/encoder.ex [hunk 1]
```diff
@@ -148,6 +148,10 @@ end
 defimpl JSON.Encoder, for: Any do
   @any_encode "[Elixir.Any]"
 
+  def encode(%{} = struct) do
+    JSON.Encoder.Helpers.dict_encode(Map.to_list(struct))
+  end
+
   def encode(_), do: JSON.Encoder.encode(@any_encode)
   def typeof(_), do: JSON.Encoder.typeof(@any_encode)
```
Comment (cblage) [inline lib/json/encoder.ex]: Can you add an overloaded `typeof` for the structs too? Otherwise it won't match

**Comment by cblage** [comment]
:100:

**Pull request merged**